azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunner.java

```diff
@@ -905,19 +911,15 @@ public void resume(String user) {
   }
 
   public void kill(String user) {
-    synchronized (mainSyncObj) {
-      logger.info("Flow killed by " + user);
-      flow.setStatus(Status.KILLED);
-      kill();
-      updateFlow();
-    }
-    interrupt();
+    logger.info("Flow killed by " + user);
+    kill();
   }
 
-  private void kill() {
+  public void kill() {
     synchronized (mainSyncObj) {
```

Comment:
Also, I would ideally wrap the entire thing in a try catch block and log stuff when we are unable to kill jobs.

Things to think about:
You have 10 jobs running. killing the 4th one failed or threw an exception. What do we do? 
 - What's the status of the flow after that?
 - How do we handle failures?